platform/engineering/design-docs/2020-04-09-separate-content-build.md

```diff
@@ -0,0 +1,498 @@
+    - Update the app generator to not add the unused properties in the
+      `manifest.json`, but log to the console with instructions on what to do to
+      make the application landing page in the content build
+    - **Estimate:** 2 - 4 days
+1. Copy the content build script to another repo
+    - The hard part here is figuring out what all the content build touches
+      - [Madge](https://github.com/pahen/madge) might be helpful for this
+    - **Estimate:** 5 - 8 days
+1. Set up the CI for the content build repo
+    - **Estimate:** ??
+    - I'm not sure what all goes into this
+        - What does it take to wire this into Jenkins / Nomad?
+1. Set up the deploy job for the content build repo
+    - **Estimate:** ??
+    - **Question:** Is this where we write a new script to copy the files from
+      the build over to S3?
+1. Once we're confident the deploys are working properly, switch the
+   `vets-website` build to build only Webpack assets
+    - Modify the deploy script to copy over only the files in `applications/`
+      - The deployment of this to Jenkins (or wherever) must be done in concert
+        with the switch to building only Webpack assets
+    - **Estimate:** 1 hour - 1 day
+1. Create the content model for application landing pages in Drupal
+    - **Estimate:** ??
+1. Create all the application landing pages in Drupal
+    - Test in staging
+    - **Estimate:** 2 hours?
+1. Create a buildtype branch in the Metalsmith script to build application
+   landing pages only in prodction
+    - So we can test the landing pages from Drupal
+    - **Estimate:** 1 hour
+1. Publish all the application landing pages and remove the `createReactPages`
+   Metalsmith step
+    - **Question:** How do we coordinate this so the cut over is clean?
+    - **Estimate:** 1 hour
+
+### Alternatives
+
+#### Mono-repo
+The alternative to splitting out the content build into its own separate repo is
+to keep it in `vets-website`. The reasons we're not doing this are:
+- To reduce what `vets-website` is "in charge" of
+  - Building static content from the CMS and compiling application code are two
+    completely separate functions
+- To keep one CI pipeline per repository
+  - And they'll be more narrowly focused
+- To manage dependencies independently
+  - All dependencies in the content build repo are not client-facing
+    - There _may_ be exceptions later
+  - It will be harder for an application to use be able to use a dependency that
+    wasn't intended to be client-facing
+
+#### Multiple S3 Buckets
+There are a few challenges that multiple buckets present.
+1. ATO
+    - There's a lot of paperwork
+1. Namespacing
+    - To use S3 to serve a static site, the bucket needs to be named after the
+      namespace, e.g. `www.VA.gov`
+    - We could maybe get around this with using a subdomain like
+      `javascript.VA.gov` or something, but again, paperwork and likely more
+      maintenance burden
```

Comment:
So actually, this is interesting.. I was originally thinking above that we could do multiple buckets and make the mapping happen in the revproxy. However, we don't even need that complexity.

Currently, the JS/CSS/etc files are served from a public S3 bucket. You'll notice that it doesn't have va.gov in the name at all. It's just a regular old S3 bucket and files are served via S3's website hosting feature. There'd be no trickery needed here and no additional paperwork. We're clear to serve this type of static content from a memo from the CTO. Technically we can also serve from a CDN as well, but we haven't gone through the vendor selection process there.

Reply:
The prod assets are currently loaded from https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com. There's no `.gov` involved 😄

Reply:
Interesting! I'd love to get some time on your calendar in the next few weeks to talk about this before we make the change to the deployments.